netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java

```diff
@@ -454,7 +478,22 @@ private static boolean installJettyTlsProtocolSelection(final SSLEngine engine,
         Class<?> serverProviderClass
             = Class.forName(protocolNegoClassName + "$ServerProvider", true, null);
         Method putMethod = negoClass.getMethod("put", SSLEngine.class, providerClass);
+
+        // When the negotiation completes, remove the extension.
         final Method removeMethod = negoClass.getMethod("remove", SSLEngine.class);
+        protocolNegotiated.addListener(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              removeMethod.invoke(null, engine);
+            } catch (IllegalAccessException e) {
```

Comment:
FYI: You can handle just ReflectiveOperationException instead of the two separate exceptions.